appservice/src/tree/DeploymentTreeItem.ts

```diff
@@ -28,15 +28,16 @@ enum DeployStatus {
 }
 
 export class DeploymentTreeItem extends AzureTreeItem<ISiteTreeRoot> {
-    public static contextValue: string = 'deployment';
-    public readonly contextValue: string = DeploymentTreeItem.contextValue;
+    public static contextValue: RegExp = new RegExp('deployment\/.*');
+    public readonly contextValue: string;
     public label: string;
     public receivedTime: Date;
     public parent: DeploymentsTreeItem;
     private _deployResult: DeployResult;
 
-    constructor(parent: DeploymentsTreeItem, deployResult: DeployResult) {
+    constructor(parent: DeploymentsTreeItem, deployResult: DeployResult, scmType?: string) {
```

Comment:
Is scmType really optional? I'm guessing this should be `scmType: string | undefined` instead. We've been over this before, but there's a difference between the two. My suggestion forces people to pass _something_ in every time they call `new DeploymentTreeItem()`. With the current implementation, it's easy to forget to pass in scmType and TypeScript won't complain